packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -255,6 +255,24 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		super.startSummary(referenceSequenceNumber, summaryLogger);
 	}
 
+	protected validateSummaryCore(parentSkipRecursion: boolean) {
+		/**
+		 * The absence of wip used routes indicates that GC was not run on this node. This can happen if:
+		 * 1. A child node was created after GC was already run on the parent. For example, a data store
+		 * is realized (loaded) after GC was run on it creating summarizer nodes for its DDSes. In this
+		 * case, the used routes of the parent should be passed on the child nodes and it should be fine.
+		 * 2. A new node was created but GC was never run on it. This can mean that the GC data generated
+		 * during summarize is complete . We should not continue, log and throw an error. This will help us
+		 * identify these cases and take appropriate action.
+		 */
+		if (!this.gcDisabled && this.wipSerializedUsedRoutes === undefined) {
+			this.throwUnexpectedError({
+				eventName: "NodeDidNotRunGC",
+			});
+		}
+		super.validateSummaryCore(parentSkipRecursion);
```

Comment:
nit: Thinking generically about an overridden `validate` function, I would do `super.validate` first, and only add on this guy's validation checks if the base class passes.

Reply:
Yeah, we can do that here. In `completeSummary`, we have to do some processing before calling super though.

Reply:
In this particular case, doing validation before calling `super.validate` makes more sense. This function validates each node in a DFS fashion so if we call `super.validate` first, it will validate that `summarize` is called on that node and then move on to calling validateSummary on child nodes without first validating GC ran for this node.
If we validate GC ran first, then for each node, valdiation for GC and summarize will happen before moving on to child nodes.
I guess it doesn't really matter because every node will get validated but it's better to validate a node completely before moving on.